text/0000-network-update-loop.md

```diff
@@ -0,0 +1,712 @@
+
+<details>
+  <summary>UML Source</summary>
+
+```
+skinparam Style strictuml
+skinparam monochrome true
+skinparam defaultFontSize 14
+
+note over PlayerLoop: Unity 2019.4 LTS
+note over NetworkUpdateLoop: RuntimeInitializeOnLoadMethod
+NetworkUpdateLoop -> NetworkUpdateLoop: Initialize
+NetworkUpdateLoop -> PlayerLoop: GetCurrentPlayerLoop
+NetworkUpdateLoop <-- PlayerLoop
+NetworkUpdateLoop -> NetworkUpdateLoop: Initialization.Add(NetworkInitialization)
+NetworkUpdateLoop -> NetworkUpdateLoop: EarlyUpdate.Insert(0, NetworkEarlyUpdate)
+NetworkUpdateLoop -> NetworkUpdateLoop: FixedUpdate.Insert(0, NetworkFixedUpdate)
+NetworkUpdateLoop -> NetworkUpdateLoop: PreUpdate.Insert(0, NetworkPreUpdate)
+NetworkUpdateLoop -> NetworkUpdateLoop: Update.Insert(0, NetworkUpdate)
+NetworkUpdateLoop -> NetworkUpdateLoop: PreLateUpdate.Insert(0, NetworkPreLateUpdate)
+NetworkUpdateLoop -> NetworkUpdateLoop: PostLateUpdate.Add(NetworkPostLateUpdate)
+NetworkUpdateLoop -> PlayerLoop: SetPlayerLoop
+NetworkUpdateLoop <-- PlayerLoop
+group Initialization
+    PlayerLoop -> PlayerLoop: PlayerUpdateTime
+    PlayerLoop -> PlayerLoop: DirectorSampleTime
+    PlayerLoop -> PlayerLoop: AsyncUploadTimeSlicedUpdate
+    PlayerLoop -> PlayerLoop: SynchronizeInputs
+    PlayerLoop -> PlayerLoop: SynchronizeState
+    PlayerLoop -> PlayerLoop: XREarlyUpdate
+    PlayerLoop -> NetworkUpdateLoop: RunNetworkInitialization
+    NetworkUpdateLoop -> NetworkUpdateLoop: AdvanceFrame
+    NetworkUpdateLoop -> NetworkUpdateLoop: ++FrameCount
+    NetworkUpdateLoop -> NetworkUpdateLoop: UpdateStage = Initialization
+    loop m_Initialization_Array
+        NetworkUpdateLoop -> INetworkUpdateSystem: NetworkUpdate
+        NetworkUpdateLoop <-- INetworkUpdateSystem
+    end
+    PlayerLoop <-- NetworkUpdateLoop
+end
+group EarlyUpdate
+    PlayerLoop -> NetworkUpdateLoop: RunNetworkEarlyUpdate
```

Comment:
This is probably too early - e.g. before profilerstartframe definitely seems wrong... maybe tie it closer to other networking needs (UnityWebRequestUpdate)